lib/components/home/artwork_rails/artwork_rail.js

```diff
@@ -35,44 +34,38 @@ class ArtworkRail extends React.Component {
     this.setState({ expanded: true })
   }
 
-  renderExpansionButton() {
-    if (!this.state.expanded) {
-      return (
-        <TouchableHighlight style={styles.expansionButton} onPress={this.expand} underlayColor={'white'}>
-            <Image style={{height: 8, width: 15, alignSelf: 'center'}} source={require('../../../../images/chevron.png')} />
-        </TouchableHighlight>
-      )
-    }
-    return null
-  }
-
-  renderViewAllButton() {
-    if (this.state.expanded) {
-      return (
-        <TouchableHighlight style={styles.viewAllButton} onPress={this.handleViewAll} underlayColor={'gray'}>
-          <Text style={styles.viewAllText}>VIEW ALL</Text>
-        </TouchableHighlight>
-      )
-    }
-    return null
+  hasAdditionalContent() {
+    const additionalContentRails = [
+      'followed_artists',
+      'saved_works',
+      'live_auctions',
+      'current_fairs',
+      'genes',
+      'generic_gene',
+    ]
+    return additionalContentRails.indexOf(this.props.rail.key) > -1
   }
 
   handleViewAll = () => {
     const context = this.props.rail.context
-    if (!context) { return }
+    const key = this.props.rail.key
 
     let url = null
-    switch (context.__typename) {
-      case 'HomePageModuleContextFollowArtists':
-        url = 'me/notifications'
+    switch (key) {
+      case 'followed_artists':
+        url = '/me/notifications'
+        break
+      case 'related_artists':
+        url = context && context.artist.href
         break
-      case 'HomePageModuleContextRelatedArtist':
-        url = context.artist.href
+      case 'saved_works':
+        url = '/user/saves#saved-artworks'
         break
```

Comment:
I think it makes sense to add a context even in those cases.

@broskoski any objections?